x-pack/plugins/ingest_manager/server/routes/agent_config/handlers.ts

```diff
@@ -37,6 +38,18 @@ export const getAgentConfigsHandler: RequestHandler<
       perPage,
       success: true,
     };
+
+    await Promise.all(
```

Comment:
Q. Should we throttle here?
From the UI, the max number of items is 50, but since this is an API I assume it could be called from elsewhere with a larger number. Not sure if there is an issue if too many concurrent `listAgents()` calls.

Reply:
Bluebird is already installed, and it provide a `map` method where you can set the concurrency, maybe we can use it here

Reply:
https://github.com/sindresorhus/p-map is also installed. How about that?

In general I'd like to stay out of they way and let the browser or something more informed manage the resources, but I agree scheduling 50 network requests does feel spammy. Maybe a number low enough to avoid meltdown but high enough to typically stay out of the way. 10?

Reply:
I don't have a preference for which lib we use. I'll go with `p-map` since its a smaller package and will use a concurrency of `10`. 

Thanks for your feedback

Reply:
FYI - So looks like `p-map` is not a top-level installed lib, but bluebird is. going with bluebird. 

```
$ npm ls p-map

├─┬ @kbn/optimizer@1.0.0
│ ├─┬ clean-webpack-plugin@3.0.0
│ │ └─┬ del@4.1.1
│ │   └── p-map@2.1.0 
│ ├─┬ cpy@8.0.0
│ │ └─┬ p-all@2.1.0
│ │   └── p-map@2.1.0  deduped
│ └─┬ terser-webpack-plugin@2.3.2
│   └─┬ cacache@13.0.1
│     └── p-map@3.0.0 
├─┬ del@5.1.0
│ └── p-map@3.0.0 
├─┬ listr@0.14.1
│ └── p-map@1.2.0 
└─┬ terser-webpack-plugin@2.3.4
  └─┬ cacache@13.0.1
    └── p-map@3.0.0 

```

Reply:
No problem. I was avoiding Bluebird based on some old memories/behavior. If we can `import { map } from 'bluebird'` I'm 👍